src/Oro/Bundle/UserBundle/Entity/UserManager.php

```diff
@@ -314,4 +326,71 @@ public function getApi(User $user, Organization $organization)
     {
         return $this->getStorageManager()->getRepository('OroUserBundle:UserApi')->getApi($user, $organization);
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function loadUserByOAuthUserResponse(UserResponseInterface $response)
+    {
+        if (!$this->cm->get('oro_user.enable_google_sso')) {
+            throw new \Exception('SSO is not enabled');
+        }
+
+        $username = $response->getUsername();
+        if ($username === null) {
+            throw new AccountNotLinkedException(sprintf("User '%s' not found.", $username));
+        }
+
+        $user = $this->findUserBy(array($this->getOAuthProperty($response) => $username));
+        if (!$user && !$this->isEmailEnabledForOauth($response->getEmail())) {
+            throw new EmailDomainNotAllowedException();
+        }
+
+        if (!$user) {
+            $user = $this->findUserByEmail($response->getEmail());
+            if ($user) {
+                $user->setGoogleId($username);
+                $this->updateUser($user);
+            }
+        }
+
+        if (!$user) {
+            throw new AccountNotLinkedException(sprintf("User '%s' not found.", $username));
+        }
+
+        return $user;
+    }
+
+    /**
+     * Returns if given email can be used for oauth
+     *
+     * @param string $email
+     *
+     * @return boolean
+     */
+    protected function isEmailEnabledForOauth($email)
+    {
+        $enabledDomain = $this->cm->get('oro_user.google_sso_domain');
+        if (!$enabledDomain) {
+            return true;
+        }
+
+        return (bool) preg_match(sprintf('/@%s$/', $enabledDomain), $email);
```

Comment:
Sure - I think it would be better (even if it would be enough in most cases to have one domain probably).

So I'll add support for multiple domains.